storage/replica_command.go

```diff
@@ -2732,7 +2745,17 @@ func (r *Replica) ChangeReplicas(
 				},
 			},
 		})
-		return txn.Run(b)
+		if err := txn.Run(b); err != nil {
+			return err
+		}
+
+		if oldDesc.RangeID != 0 && !reflect.DeepEqual(oldDesc, desc) {
```

Comment:
@petermattis Because it's only here that the transaction has committed. Conceivably we might try to change replicas and the CPut would fail because of a race.

@BramGruneir this is pretty far off the hot execution path.